vslib/src/TrafficForwarder.cpp

```diff
@@ -0,0 +1,100 @@
+#include "TrafficForwarder.h"
+
+#include "swss/logger.h"
+
+#include <memory.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <net/if.h>
+#include <linux/if_tun.h>
+#include <sys/ioctl.h>
+#include <net/if_arp.h>
+#include <unistd.h>
+#include <net/ethernet.h>
+#include <arpa/inet.h>
+#include <linux/if_packet.h>
+#include <linux/if_ether.h>
+
+using namespace saivs;
+
+#define ETH_FRAME_BUFFER_SIZE (0x4000)
+#define CONTROL_MESSAGE_BUFFER_SIZE (0x1000)
+#define IEEE_8021Q_ETHER_TYPE (0x8100)
+#define MAC_ADDRESS_SIZE (6)
+#define VLAN_TAG_SIZE (4)
+
+void TrafficForwarder::add_vlan_tag(
+    _Inout_ unsigned char *buffer,
+    _Inout_ size_t &length,
```

Comment:
so there is a potential problem here, which also exists in original code, so buffer is allocated with ETH_FRAME_BUFFER_SIZE, but length here is specifying valid data in the buffer, which probably all the time is less then buffer allocated size, but since buffer is extended here, and vlan tag is inserted, then in case when length will be exact == ETH_FRAME_BUFFER_SIZE, then write will occur outside buffer bounds, which can cause issues. maybe we could add check for that ? and throw before buffer is modified ?

Reply:
Done, please check it.